languages/go/reference/exercise-concepts/hamming.md

```diff
@@ -0,0 +1,29 @@
+# Hamming
+
+## Exercise
+
+Calculate the Hamming Distance between two DNA strands.
+
+## Concepts encountered while solving it
+
+## Strings
+Strings in Go are made up of runes. Runes are Go's equivalent to the concept of characters. A rune can be represented using from 1 up to 4 bytes. 
+
+### String iteration
+- When iterating over strings using the range operator, runes are yielded. When dealing with ASCII runes, strings can be safely ranged using the string[i] syntax. However when non ASCII characters are present this is unsafe as only one byte of the corresponing rune is returned.
+
+### String comparison
+- Since two strings cannot be ranged using the range operator at the same time, the safe approach for comparing each "character" in two strings is by first casting the strings to a slice of runes.
+
+## Errors
+In Go there are no exceptions, instead errors can be defined and processed accordingly to the context. Errors can be defined using fmt.Errorf or errors.New. The latter is preferred when there is no string interpolation within the error message. Error messages should be concise, not vague and provide context for where and why the error occurred. Also they should not be capitalized or have punctuation.
+
+When an error is encountered the convention is to return a zero value along with the corresponging error. In the opposite case a nil error should be returned along with the value that the function should return.
```

Comment:
Maybe split this up in 2 concepts: `error creation` and `returning errors`. Those 2 are something we should probably teach very early on in a concept exercise maybe called `error-basics`.

Reply:
To `returning errors` also belongs that they should always be the last return parameter. Doesn't need to mentioned here. Just mentioning it as there are a few things to teach here.

Reply:
I like the idea of having an early exercise with basic error concepts, given how common they are.